packages/shared/src/components/plus/PlusProductList.tsx

```diff
@@ -0,0 +1,138 @@
+import type { ReactElement } from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import type { ProductOption } from '../../contexts/PaymentContext';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../typography/Typography';
+import { RadioItem } from '../fields/RadioItem';
+import { LogEvent } from '../../lib/log';
+import { usePlusSubscription } from '../../hooks';
+import { PlusPriceType } from '../../lib/featureValues';
+import { feature } from '../../lib/featureManagement';
+import { useFeature } from '../GrowthBookProvider';
+
+type PlusProductListProps = {
+  productList: ProductOption[];
+  selected?: string;
+  onChange: (value: string) => void;
+  backgroundImage?: string;
+  className?: string;
+};
+
+const PlusProductList = ({
+  productList,
+  selected,
+  onChange,
+  className,
+}: PlusProductListProps): ReactElement => {
+  const { logSubscriptionEvent } = usePlusSubscription();
+  const planTypes = useFeature(feature.pricingIds);
+
+  const earlyAdopterPlanId = useMemo(() => {
+    const monthlyPrices = productList.filter(
+      (option) => planTypes[option.value] === PlusPriceType.Monthly,
+    );
+
+    if (monthlyPrices.length <= 1) {
+      return null;
+    }
+
+    return monthlyPrices.reduce((acc, plan) => {
+      return acc.priceUnformatted < plan.priceUnformatted ? acc : plan;
+    }).value;
+  }, [productList, planTypes]);
```

Comment:
Don't we abstract this somewhere already?
If we do this more then once, consider putting it in hook/common function?